standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HMSHandler.java

```diff
@@ -1363,18 +1363,62 @@ private void create_database_core(RawStore ms, final Database db)
   }
 
   @Override
+  @Deprecated
   public void create_database(final Database db)
       throws AlreadyExistsException, InvalidObjectException, MetaException {
-    startFunction("create_database", ": " + db.toString());
+    CreateDatabaseRequest req = new CreateDatabaseRequest(db.getName());
+    if (db.isSetDescription()) {
```

Comment:
if the description is null, it's safe to call req.setDescription(null), we need not an extra db.isSetDescription()?

Reply:
A CreateDatabaseRequest.Builder here to wrap the attributes is better.

Reply:
There is no builder class in the CreateDatabaseRequest thrift object.